### PR TITLE
CSV layer

### PR DESCRIPTION
**Pull request by jb55**
CSV layer for Spacemacs!

Ported from my spacemacs-csv, but without the extension since csv-mode doesn't seem to have modeline issues anymore.

### CODE REVIEW

**Comment by puffnfresh** [comment]
:+1:

File: layers/csv/packages.el [hunk 1]
```diff
@@ -0,0 +1,24 @@
+(defvar csv-packages
```
Comment (person808) [inline layers/csv/packages.el]: Should use `setq` so `SPC f e R` works. Also need to remove docstring

Comment (jb55) [inline layers/csv/packages.el]: kk fixing

**Comment by StreakyCobra** [comment]
Just asking, wouldn't it be better for such simple package to put it in `extra-lang` ? Also, `yaml` is in `+lang`, I suppose `CSV` can be here too, maybe.

**Comment by jb55** [comment]
csv isn't a language tho

File: layers/csv/packages.el [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(setq csv-packages '(csv-mode))
+
+(defun csv/init-csv-mode ()
+  "Initial csv mode"
+  (use-package csv-mode
+    :init
+    (evil-leader/set-key-for-mode 'csv-mode
```
Comment (TheBB) [inline layers/csv/packages.el]: Please use `spacemacs/set-leader-keys-for-mode`

**Comment by TheBB** [comment]
Please also use the updated README.org template. You have some things missing.

**Comment by TheBB** [comment]
I agree we can move probably move it to `+lang`.

**Comment by jb55** [comment]
ok updating

**Comment by jb55** [comment]
updated to template, do I need an image though? Not sure what I would use for csv.

**Comment by TheBB** [comment]
Excellent :+1:. No, you don't need an image.

**Comment by CarlQLange** [comment]
:+1: Can't wait for this

**Comment by robbyoconnor** [comment]
:heart:

**Comment by jb55** [comment]
:wave: friendly bump against the avalanche of PRs 💫

**Comment by CarlQLange** [comment]
I've pulled this locally and used it for months now. I deal with CSVs probably 5 times a day. Merge this, it's perfect!!

**Comment by syl20bnr** [comment]
Bump registered :-)

**Comment by robbyoconnor** [comment]
This has been cherrypicked and @syl20bnr will likely say you can delete your branch safely now ;) :tongue:

**Comment by syl20bnr** [comment]
Thank you :+1: 
I made some minor improvements in the commits following this cherry-picked one.
You can indeed safely delete your branch :-)

**Pull request closed without merging**